### DISCUSSION TITLE
Missing docs regarding user traits.

### DISCUSSION THREAD

**Discussion by tooboredtocode** [gravitational/teleport#7200] [Q&A]
Regarding the PR that adds the feature of rewriting headers (#6601), how do I get access to the users username? So I can map teleport users to grafana users?
I haven't found anything regarding this in the documentation.

Any help is greatly appreciated

**Comment by webvictim** [comment]
There are some docs here which might help: https://goteleport.com/docs/access-controls/reference/

Essentially I don't think you can use anything other than `internal.logins` or `external.<trait_name>` currently. I don't know whether `internal.<other_trait_name>` will work. You can use `tsh -d status` once logged in to see a list of traits set against your user.

**Reply by tooboredtocode** [reply]
Is there any way to get access to the username?

**Reply by webvictim** [reply]
Not that I know of. If it appears under traits in `tsh -d status` you should be able to use it, if not I don't think there's a way to do it.

**Reply by ahelwer** [reply]
I've been experimenting and I think only specific traits are accessible via `{{internal.traitname}}`; see [this question](https://github.com/gravitational/teleport/discussions/8878#discussion-3669479).
